File: 2020/13.roman-to-integer.js
/*
 * @lc app=leetcode id=13 lang=javascript
 *
 * [13] Roman to Integer
 */

// @lc code=start
/**
 * @param {string} s
 * @return {number}
 */
var romanToInt = function(roman) {
  var data = {M: 1000, D: 500, C: 100, L: 50, X: 10, V: 5, I: 1};
  var numbers = roman.split('');
  var sum = 0, i;

  for(i = 0; i < numbers.length; i++)
  {
    if(data[numbers[i]] < data[numbers[i+1]])
    {
      sum += data[numbers[i+1]] - data[numbers[i]];
      i++;
    }
    else
    {
      sum += data[numbers[i]];
    }
  }
  
  return sum;
};
// @lc code=end

File: 13.roman-to-integer.py
#
# @lc app=leetcode id=13 lang=python3
#
# [13] Roman to Integer
#

# @lc code=start
class Solution:
    def romanToInt(self, s: str) -> int:
        v = {'count': 0, 's': s}

        self.check_remove(v, 'IV', 4)
        self.check_remove(v, 'IX', 9)
        self.check_remove(v, 'XL', 40)
        self.check_remove(v, 'XC', 90)
        self.check_remove(v, 'CD', 400)
        self.check_remove(v, 'CM', 900)

        self.check_remove(v, 'I', 1)
        self.check_remove(v, 'V', 5)
        self.check_remove(v, 'X', 10)
        self.check_remove(v, 'L', 50)
        self.check_remove(v, 'C', 100)
        self.check_remove(v, 'D', 500)
        self.check_remove(v, 'M', 1000)

        return v['count']
        
    def check_remove(self, v, text, value):
        if text in v['s']:
            print(v)
            print(text)
            v['count'] += v['s'].count(text) * value
            v['s'] = v['s'].replace(text, '')

# @lc code=end
